### PR TITLE
Telemetry scalar for the bandwidth

### PR DESCRIPTION
**Pull request by bakulf**

### CODE REVIEW

File: docs/metrics.md [hunk 1]
```diff
@@ -111,6 +114,33 @@ The **methods** are:
...
 
+### scalar Registration and Recording
+
+Scalars are defined by **registering** them using a call to `browser.telemetry.registerScalars(category, scalarData)`.
+
+Here's a breakdown of how a to register a typical scalar:
+
+
+```javascript
+browser.telemetry.registerScalars("eventcategory", {
```
Comment (chutten) [inline docs/metrics.md]:
```suggestion
browser.telemetry.registerScalars("scalar.category", {
```

File: docs/metrics.md [hunk 2]
```diff
@@ -111,6 +114,33 @@ The **methods** are:
...
+    "scalartName": {
```
Comment (chutten) [inline docs/metrics.md]:
```suggestion
    "scalar_name": {
```

File: docs/metrics.md [hunk 3]
```diff
@@ -111,6 +114,33 @@ The **methods** are:
...
+        kind: COUNT // the type of the scalar
```
Comment (chutten) [inline docs/metrics.md]: You might need to specify that this must match a constant on `nsITelemetry`

File: docs/metrics.md [hunk 4]
```diff
@@ -111,6 +114,33 @@ The **methods** are:
...
+    }
+```
+
+Once an scalar is registered, we can add values to it with: (\*)
+
+`browser.telemetry.scalarAdd(category, value)`
```
Comment (chutten) [inline docs/metrics.md]:
```suggestion
`browser.telemetry.scalarAdd("scalar.category.scalar_name", value)`
```

File: docs/metrics.md [hunk 5]
```diff
@@ -111,6 +114,33 @@ The **methods** are:
    1. `exemptTab`: the user has decided to disable the proxy for this tab.
 
 
+### scalar Registration and Recording
```
Comment (chutten) [inline docs/metrics.md]: For this section you might just wish to link out to the [in-tree Scalar docs](https://firefox-source-docs.mozilla.org/toolkit/components/telemetry/collection/scalars.html) and force us to keep them updated for you : )

File: src/background/main.js [hunk 6]
```diff
@@ -579,9 +579,12 @@ class Main {
       case "excludedDomains":
         return this.fxa.excludedDomains();
 
-      case "telemetry":
+      case "telemetryEvent":
         return this.telemetry.syncAddEvent(data.category, data.event);
 
+      case "telemetryScalar":
+        return this.telemetry.syncAddScalar(data.category, data.value);
```
Comment (chutten) [inline src/background/main.js]: It's not a category, it's the scalar name minus the scalar category

File: src/background/network.js [hunk 7]
```diff
@@ -473,4 +480,31 @@ export class Network extends Component {
       this.proxyPassthrough.add(host.trim());
     });
   }
+
+  async maybeSendTelemetryBandwidthEvent(details) {
```
Comment (chutten) [inline src/background/network.js]: It's not an event, maybe "message"?

File: src/background/telemetry.js [hunk 8]
```diff
@@ -73,4 +85,13 @@ export class Telemetry extends Component {
       console.error("Telemetry.recordEvent failed", e);
     });
   }
+
+  syncAddScalar(category, value) {
+    log(`Sending telemetry scalar: ${category} - ${value}`);
+
+    // eslint-disable-next-line verify-await/check
+    browser.telemetry.scalarAdd(TELEMETRY_CATEGORY + "." + category, value).catch(e => {
```
Comment (chutten) [inline src/background/telemetry.js]: I think we want to call it `scalarName` instead of `category`

File: src/background/main.js (hunk 6 above)
Comment (bakulf) [reply src/background/main.js]: Can I call it "key" ?

Comment (chutten) [reply src/background/main.js]: Only if you never use keyed scalars : )

**Comment by bakulf** [comment]
> Looks fine from a Telemetry POV (still needs Data Collection Review). You may also wish to have someone who knows secure-proxy code take a look.

Data-review on bugzilla, right?

**Comment by chutten** [comment]
> 
> 
> > Looks fine from a Telemetry POV (still needs Data Collection Review). You may also wish to have someone who knows secure-proxy code take a look.
> 
> Data-review on bugzilla, right?

That's where all the infrastructure is for flagging reviews to stewards and sending emails and doing dashboarding and stuff. We've done reviews in other places, but I'm never completely happy when it's not on a mozilla.org domain : )

**Pull request merged**